### PR TITLE
fix(core): Remove threads pkg, rewrite log writer worker

### PR DESCRIPTION
**Pull request by flipswitchingmonkey**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts [hunk 1]
```diff
@@ -210,16 +216,13 @@ class MessageEventBus extends EventEmitter {
...
+		// LoggerProxy.debug(`Listeners: ${this.eventNames().join(',')}`);
 
-		// if there are no set up destinations, immediately mark the event as sent
 		if (
-			!isLogStreamingEnabled() ||
-			Object.keys(this.destinations).length === 0 ||
-			!this.hasAnyDestinationSubscribedToEvent(msg)
+			isLogStreamingEnabled() &&
+			Object.keys(this.destinations).length > 0 &&
+			this.hasAnyDestinationSubscribedToEvent(msg)
```
Comment (netroy) [inline packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts:224]: we have this almost identical check in `send`. we could extract this into a private function 
```typescript
	private shouldSend(msg: EventMessageTypes): boolean {
		return (
			isLogStreamingEnabled() &&
			Object.keys(this.destinations).length > 0 &&
			this.hasAnyDestinationSubscribedToEvent(msg)
		);
	}
```

and update this to 
```suggestion
		shouldSend(msg)
```

File: packages/cli/src/eventbus/MessageEventBusWriter/MessageEventBusLogWriter.ts [hunk 2]
```diff
@@ -73,50 +79,52 @@ export class MessageEventBusLogWriter {
...
 	}
 
 	private async spawnThread(): Promise<boolean> {
-		this.worker = await spawn<MessageEventBusLogWriterWorker>(
-			new Worker(`${parse(__filename).name}Worker`),
-		);
+		const parsedName = parse(__filename);
+		let workerFileName;
+		if (inTest) {
+			workerFileName = './dist/eventbus/MessageEventBusWriter/MessageEventBusLogWriterWorker.js';
```
Comment (netroy) [inline packages/cli/src/eventbus/MessageEventBusWriter/MessageEventBusLogWriter.ts:111]: how is `threads` resolving the worker file name in tests?

File: packages/cli/src/eventbus/MessageEventBusWriter/MessageEventBusLogWriter.ts [hunk 3]
```diff
@@ -73,50 +79,52 @@ export class MessageEventBusLogWriter {
...
 				await MessageEventBusLogWriter.instance.startThread();
 			});
 			return true;
 		}
 		return false;
 	}
 
-	getThread(): ModuleThread<MessageEventBusLogWriterWorker> | undefined {
+	getWorker(): Worker | undefined {
 		if (this.worker) {
```
Comment (netroy) [inline packages/cli/src/eventbus/MessageEventBusWriter/MessageEventBusLogWriter.ts:128]: why not just `return this.worker` ?

File: packages/cli/src/eventbus/MessageEventBusWriter/MessageEventBusLogWriter.ts (hunk 2 above)
Comment (flipswitchingmonkey) [reply packages/cli/src/eventbus/MessageEventBusWriter/MessageEventBusLogWriter.ts:111]: threads transpiles it via ts-node and then figures out the path using require.resolve. see https://github.com/andywer/threads.js/blob/master/src/master/implementation.node.ts

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.211.0`

**Pull request merged**